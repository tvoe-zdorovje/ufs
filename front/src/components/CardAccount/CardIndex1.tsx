import React from 'react'
import {
  Row,
  Col,
  Button
} from 'antd'
import renderPair from './RenderPair'

const styles = require('./CardAccount.less')

export interface CardIndex1Props {
  items: Array<any>,
  addSidePanel: Function,
}

export default function CardIndex1({ items, addSidePanel }: CardIndex1Props) {
  const listItems = items.map((item, i) => {
    const pairs = [['Номер документа', item.docId],
      ['Вид', item.docType],
      ['Получатель', item.recipientShortName],
      ['Сумма', item.amount],
      ['Дата', item.comeInDate]];
    return (
      <Row className={styles.separatedRow} key={i}>
        <Col span={23}>
          {pairs.map((pair, j) => renderPair(pair[0], pair[1], j))}
        </Col>
        <Col span={1}>
          <Button
            className={styles.largeIcon}
            onClick={() => addSidePanel(pairs, `ПП ${item.docId}`)}
            icon="menu-unfold"
          />
        </Col>
      </Row>
    )
  })

  return (
    <div>
      {listItems}
    </div>
  )
}
