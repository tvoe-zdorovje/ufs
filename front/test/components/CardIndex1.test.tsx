import React from 'react'
import {
  render,
  shallow
} from 'enzyme'
import toJson from 'enzyme-to-json'
import { Button } from 'antd'

import CardIndex1, { CardIndex1Props } from 'components/CardAccount/CardIndex1'

describe('Компонент CardIndex1', () => {

  const props: CardIndex1Props = {
    items: [{
      docId: '98710245', docType: 'Платежное поручение', recipientShortName: 'Газпром', amount: '20000',
      comeInDate: '01.05.2017'
    }, {
      docId: '98710246', docType: 'Платежное поручение', recipientShortName: 'Газпром', amount: '20000',
      comeInDate: '01.05.2017'
    }],
    addSidePanel: (x, y) => {
    },
  }

  const renderWrapper = render(<CardIndex1 {...props} />)
  const shallowWrapper = shallow(<CardIndex1 {...props} />)

  it('выполняется рендер всех item', () => {
    expect(shallowWrapper.children()).toHaveLength(2)
  })

  it('работает кнопка', () => {
    const spy = jest.fn()
    shallowWrapper.setProps({ addSidePanel: spy })
    shallowWrapper.update()
    shallowWrapper.find(Button).at(0).simulate('click', {
      preventDefault() {
      }
    })
    expect(spy).toBeCalled()
  })

  it('совпадает со своим снимком', () => {
    expect(toJson(renderWrapper)).toMatchSnapshot()
  })
})
