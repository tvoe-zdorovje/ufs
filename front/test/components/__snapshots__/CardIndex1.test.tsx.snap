// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент CardIndex1 совпадает со своим снимком 1`] = `
<div>
  <div
    class="ant-row separatedRow"
  >
    <div
      class="ant-col-23"
    >
      <div
        class="ant-row pairRow"
      >
        <div
          class="ant-col-8 rightAligned bold"
        >
          Номер документа
        </div>
        <div
          class="ant-col-16"
        >
          98710245
        </div>
      </div>
      <div
        class="ant-row pairRow"
      >
        <div
          class="ant-col-8 rightAligned bold"
        >
          Вид
        </div>
        <div
          class="ant-col-16"
        >
          Платежное поручение
        </div>
      </div>
      <div
        class="ant-row pairRow"
      >
        <div
          class="ant-col-8 rightAligned bold"
        >
          Получатель
        </div>
        <div
          class="ant-col-16"
        >
          Газпром
        </div>
      </div>
      <div
        class="ant-row pairRow"
      >
        <div
          class="ant-col-8 rightAligned bold"
        >
          Сумма
        </div>
        <div
          class="ant-col-16"
        >
          20000
        </div>
      </div>
      <div
        class="ant-row pairRow"
      >
        <div
          class="ant-col-8 rightAligned bold"
        >
          Дата
        </div>
        <div
          class="ant-col-16"
        >
          01.05.2017
        </div>
      </div>
    </div>
    <div
      class="ant-col-1"
    >
      <button
        class="ant-btn ant-btn-icon-only largeIcon"
        type="button"
      >
        <i
          class="anticon anticon-menu-unfold"
        />
      </button>
    </div>
  </div>
  <div
    class="ant-row separatedRow"
  >
    <div
      class="ant-col-23"
    >
      <div
        class="ant-row pairRow"
      >
        <div
          class="ant-col-8 rightAligned bold"
        >
          Номер документа
        </div>
        <div
          class="ant-col-16"
        >
          98710246
        </div>
      </div>
      <div
        class="ant-row pairRow"
      >
        <div
          class="ant-col-8 rightAligned bold"
        >
          Вид
        </div>
        <div
          class="ant-col-16"
        >
          Платежное поручение
        </div>
      </div>
      <div
        class="ant-row pairRow"
      >
        <div
          class="ant-col-8 rightAligned bold"
        >
          Получатель
        </div>
        <div
          class="ant-col-16"
        >
          Газпром
        </div>
      </div>
      <div
        class="ant-row pairRow"
      >
        <div
          class="ant-col-8 rightAligned bold"
        >
          Сумма
        </div>
        <div
          class="ant-col-16"
        >
          20000
        </div>
      </div>
      <div
        class="ant-row pairRow"
      >
        <div
          class="ant-col-8 rightAligned bold"
        >
          Дата
        </div>
        <div
          class="ant-col-16"
        >
          01.05.2017
        </div>
      </div>
    </div>
    <div
      class="ant-col-1"
    >
      <button
        class="ant-btn ant-btn-icon-only largeIcon"
        type="button"
      >
        <i
          class="anticon anticon-menu-unfold"
        />
      </button>
    </div>
  </div>
</div>
`;
